gh-101467: Correct py.exe handling of prefix matches and cases when only one runtime is installed (#101468)
python/cpython: Doc/using/windows.rst

  py -2

If you see the following error, you do not have the launcher installed::

  'py' is not recognized as an internal or external command,
  operable program or batch file.

The command::

  py --list

displays the currently installed version(s) of Python.

The ``-x.y`` argument is the short form of the ``-V:Company/Tag`` argument,
which allows selecting a specific Python runtime, including those that may have
come from somewhere other than python.org. Any runtime registered by following

> :pep:`514` will be discoverable. The ``py -0`` command lists all available

Author: ```suggestion
:pep:`514` will be discoverable. The ``--list`` command lists all available
```

To match the above example

runtimes using the ``-V:`` format.

When using the ``-V:`` argument, specifying the Company will limit selection to
runtimes from that provider, while specifying only the Tag will select from all
providers. Note that omitting the slash implies a tag::

>   # Select any '3.9' tagged runtime
>   py -V:3.9

Author: ```suggestion
  # Select any '3.*' tagged runtime
  py -V:3
```

  # Select any 'PythonCore' released runtime
  py -V:PythonCore/

  # Select PythonCore's latest Python 3 runtime
  py -V:PythonCore/3

The short form of the argument (``-3``) only ever selects from core Python
releases, and not other distributions. However, the longer form (``-V:3``) will
select from any.

The Company is matched on the full string, case-insenitive. The Tag is matched
oneither the full string, or a prefix, provided the next character is a dot or a
hyphen. This allows ``-V:3.1`` to match ``3.1-32``, but not ``3.10``. Tags are

> sorted using numerical ordering (``3.10`` is 'bigger' than ``3.1``), but are not
> compared using it (``-V:3.01`` will not match ``3.1``).

Author: ```suggestion
sorted using numerical ordering (``3.10`` is newer than ``3.1``), but are
compared using text (``-V:3.01`` does not match ``3.1``).
```
